### PR TITLE
Move to seurat 3.2.3

### PR DESCRIPTION
**Pull request by pcm32**
Moves to Seurat 3.2.3, mostly to:
- Introduce seurat converter 
- Leave a reasonable IO layer on seurat 3 before moving to Seurat 4
- Switch defaults to the ones used in Seurat 3.2.3 in some detected cases, and changing parameter names when this applied.

After this upgrade, I only intend to introduce for Seurat 3 a single galaxy tool for format conversion.

### CODE REVIEW

**Comment by pcm32** [comment]
@a-solovyev12 do you might have some data that would allow to test the seurat integration method? Thanks!

**Comment by pcm32** [comment]
I'm opening a chance to review this since it is in a group repo, but I don't think it is strictly required if you just want to skim this (it is fairly long). If there is no review in a coupled days, which is perfectly fine, I'll probably abuse of my privilege. All tests passing of course. After merging this, I intend to merge #29 which have been sort of developed in parallel, with some syncing here or there, but ideally not a lot of additional changes should be introduced here or that will make syncing more complex.

File: seurat-convert.R [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    default = "seurat",
+    type = 'character',
+    help = "Either loom, seurat, anndata or singlecellexperiment for the input format to read."
+  ),
+  make_option(
+    c("--output-format"),
+    action = "store",
+    default = "seurat",
+    type = 'character',
+    help = "Either loom, seurat, anndata or singlecellexperiment for the output format."
```
Comment (pmb59) [inline seurat-convert.R:33]: to avoid misunderstanding, the values for the different options could be quoted

**Review by pmb59 — APPROVED** [review]

**Comment by pcm32** [comment]
Thanks @pmb59 for the review.

**Pull request merged**